src/plus/gk/checkin.ts

```diff
@@ -176,10 +186,17 @@ export function getSubscriptionFromCheckIn(
 			new Date(license.latestEndDate),
 			license.latestStatus === 'cancelled',
 			license.nextOptInDate ?? data.nextOptInDate,
+			(license.latestStatus === 'in_trial' || license.latestStatus === 'trial') &&
+				latestEndDate > today &&
+				today > latestStartDate,
 		);
 	}
 
-	if (effective == null || getSubscriptionPlanPriority(actual.id) >= getSubscriptionPlanPriority(effective.id)) {
+	if (
+		effective == null ||
+		(getSubscriptionPlanPriority(actual.id) >= getSubscriptionPlanPriority(effective.id) &&
+			!isSubscriptionExpired(actual))
```

Comment:
I'm not sure how this passed the build check on the PR, but `isSubscriptionExpired` is expecting a different type than it is being sent here.

![image](https://github.com/user-attachments/assets/04b92cc2-f96a-43f6-87cc-97039d6e8183)

Maybe you can just check the expiration date directly on it since its a `SubscriptionPlan`?

Reply:
You're right, not sure why it didn't error before 🤔 
Fixed it, thanks for the catch!